rcljava/src/main/java/org/ros2/rcljava/qos/QoSProfile.java

```diff
@@ -15,21 +15,36 @@
 
 package org.ros2.rcljava.qos;
 
+import java.time.Duration;
+
 import org.ros2.rcljava.qos.policies.Durability;
 import org.ros2.rcljava.qos.policies.History;
+import org.ros2.rcljava.qos.policies.Liveliness;
 import org.ros2.rcljava.qos.policies.QoSPolicy;
 import org.ros2.rcljava.qos.policies.Reliability;
 
 public class QoSProfile {
-  private final History history;
+  private History history;
+
+  private int depth;
+
+  private Reliability reliability;
+
+  private Durability durability;
 
-  private final int depth;
+  private Duration deadline = Duration.ofSeconds(0, 0);
 
-  private final Reliability reliability;
+  private Duration lifespan = Duration.ofSeconds(0, 0);
 
-  private final Durability durability;
+  private Liveliness liveliness = Liveliness.SYSTEM_DEFAULT;
 
-  private final boolean avoidROSNamespaceConventions;
+  private Duration livelinessLeaseDuration = Duration.ofSeconds(0, 0);
+
+  private boolean avoidROSNamespaceConventions;
+
+  public QoSProfile(int depth) {
+    this(History.KEEP_LAST, depth, Reliability.RELIABLE, Durability.VOLATILE, false);
+  }
 
   public QoSProfile(History history, int depth, Reliability reliability, Durability durability,
```

Comment:
What do you think about adding deadline, liveliness, and lifespan to this constructor? Then we can apply the default values for them in the new single-parameter constructor above.

Reply:
I actually wanted to deprecate this constructor, which is not quite ergonomic.
I added default values together with the definition of the new member variable.

Reply:
Sure, I think we'll have to propose it's deprecation upstream later.
I'd rather not diverge too much.